ddtrace/_trace/tracing_plugins/base/consumer.py

```diff
@@ -0,0 +1,132 @@
+"""
+ConsumerPlugin - Base for message consumers.
+
+Extends InboundPlugin since consuming messages is an inbound operation
+(receiving data FROM a message broker/queue).
+"""
+
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Optional
+from typing import Tuple
+
+from ddtrace._trace.tracing_plugins.base.inbound import InboundPlugin
+
+
+if TYPE_CHECKING:
+    from ddtrace._trace.span import Span
+    from ddtrace._trace.tracing_plugins.base.events import MessagingContext
+    from ddtrace.internal.core import ExecutionContext
+
+
+class ConsumerPlugin(InboundPlugin):
+    """
+    Base plugin for message consumers.
+
+    Extends InboundPlugin - consuming is receiving data IN from a broker.
+
+    Handles:
+    - Trace context extraction from message headers
+    - Span linking to producer span
+    - Destination tagging (topic, queue, exchange)
+    - Message metadata tagging (offset, partition, etc.)
+
+    Subclasses define:
+    - package: str (e.g., "kafka", "rabbitmq")
+    - operation: str (e.g., "consume", "receive")
+    """
+
+    from ddtrace.ext import SpanKind
+
+    kind = SpanKind.CONSUMER
+    type = "worker"
+
+    def on_start(self, ctx: "ExecutionContext") -> None:
+        """
+        Create consumer span with context extraction.
+
+        Reads from ctx:
+        - pin: The Pin instance
+        - messaging_context: MessagingContext with destination, headers, etc.
+        - span_name: Optional override for span name
+        """
+        from ddtrace.constants import _SPAN_MEASURED_KEY
+        from ddtrace.ext import SpanTypes
+        from ddtrace.propagation.http import HTTPPropagator
+
+        pin = ctx.get_item("pin")
+        msg_ctx: Optional["MessagingContext"] = ctx.get_item("messaging_context")
+
+        if not pin or not pin.enabled() or not msg_ctx:
+            return
+
+        # Determine span name
+        span_name = ctx.get_item("span_name") or f"{msg_ctx.messaging_system}.consume"
+
+        # Create span
+        span = self.start_span(
+            ctx,
+            span_name,
+            resource=msg_ctx.destination,
+            span_type=SpanTypes.WORKER,
+        )
+
+        if not span:
+            return
+
+        # Mark as measured
+        span.set_metric(_SPAN_MEASURED_KEY, 1)
```

Comment:
This is a very important tag for trace metrics. Now that we are getting rid of integration specific service names we will ONLY compute trace metrics for spans with this tag OR tags that have a span.kind of producer, server, client, and producer.

To be safe we should move this to be a core component for all plugins. This should be set on the first span for each integration